### PR TITLE
For #10756 -  UI intermittent media playback failures

### LINKED ISSUE #10756: FNX-4531 ⁃ [Bug] Intermittent UI test error: videoPlaybackSystemNotificationTest
Firebase: https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6337539131377699799/executions/bs.6606eaebeafc7818/testcases/1/test-cases

```
 androidx.test.uiautomator.UiObjectNotFoundException: for text= "Video_Test_Page"
at androidx.test.uiautomator.UiCollection.getChildByText(UiCollection.java:126)
at androidx.test.uiautomator.UiScrollable.getChildByText(UiScrollable.java:215)
at org.mozilla.fenix.ui.robots.NotificationRobot.verifySystemNotificationExists(NotificationRobot.kt:28)
at org.mozilla.fenix.ui.MediaNotificationTest$videoPlaybackSystemNotificationTest$3.invoke(MediaNotificationTest.kt:57)
at org.mozilla.fenix.ui.MediaNotificationTest$videoPlaybackSystemNotificationTest$3.invoke(MediaNotificationTest.kt:27)
at org.mozilla.fenix.ui.robots.BrowserRobot$Transition.openNotificationShade(BrowserRobot.kt:350)
at org.mozilla.fenix.ui.MediaNotificationTest.videoPlaybackSystemNotificationTest(MediaNotificationTest.kt:56) 
```

### PR DESCRIPTION
**Pull request by isabelrios**
Trying to fix #10756 and related issues.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 1]
```diff
@@ -51,8 +51,10 @@ class MediaNotificationTest {
 
         navigationToolbar {
         }.enterURLAndEnterToBrowser(videoTestPage.url) {
+            verifyPageContent("Page content: video player")
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt:54]: I think we can check against `videoTestPage.content.

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 2]
```diff
@@ -161,5 +165,6 @@ class MediaNotificationTest {
 
         // close notification shade before the next test
         mDevice.pressBack()
+        homeScreen { }.togglePrivateBrowsingMode()
     }
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt:169]: I would add a comment why this is needed.  This looks fine to me if this ensures the notification shade is closed.

**Comment by AaronMT** [comment]
(@sv-ohorvath feel free to look this over too)

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt (hunk 1 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt:54]: Ok, will  change that

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt (hunk 2 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt:169]: yes, I have seen in the failures video that the next test starts on private mode while it should not, trying to reduce  the failure if we start in regular mode as expected

File: app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt [hunk 3]
```diff
@@ -51,8 +51,10 @@ class MediaNotificationTest {
 
         navigationToolbar {
         }.enterURLAndEnterToBrowser(videoTestPage.url) {
+            verifyPageContent("Page content: video player")
             clickMediaPlayerPlayButton()
             waitForPlaybackToStart()
+            verifyPageContent("Media file is playing")
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/MediaNotificationTest.kt:57]: waitForPlaybackToStart() verifies the same thing, maybe you can re-write that method if needed or remove it

**Pull request merged**